src/editingcontroller.js

```diff
@@ -6,12 +6,157 @@
 'use strict';
 
 import ViewDocument from './view/document.js';
+import MutationObserver from './view/observer/mutationobserver.js';
+import SelectionObserver from './view/observer/selectionobserver.js';
+import FocusObserver from './view/observer/focusobserver.js';
+import KeyObserver from './view/observer/keyobserver.js';
 
+import Mapper from './conversion/mapper.js';
+import ModelConversionDispatcher from './conversion/modelconversiondispatcher.js';
+import { insertText, remove, move } from './conversion/model-to-view-converters.js';
+import { convertSelectionChange } from './conversion/view-selection-to-model-converters.js';
+import {
+	convertRangeSelection,
+	convertCollapsedSelection,
+	clearAttributes
+} from './conversion/model-selection-to-view-converters.js';
+
+import EmitterMixin from '../utils/emittermixin.js';
+
+/**
+ * Controller for the editing pipeline. The editing pipeline controls {@link engine.EditingController#model model} rendering,
+ * including selection handling. It also creates {@link engine.EditingController#view view document} which build a
+ * browser-independent virtualization over the DOM elements. Editing controller also attach default converters and
+ * observers.
+ *
+ * Note that the following observers are attached by the controller and are always available:
+ * * {@link view.observer.MutationObserver},
+ * * {@link view.observer.SelectionObserver},
+ * * {@link view.observer.FocusObserver},
+ * * {@link view.observer.KeyObserver}.
+ *
+ * @memberOf engine
+ */
 export default class EditingController {
-	constructor( modelDocument ) {
-		this.model = modelDocument;
+	/**
+	 * Creates editing controller instance.
+	 *
+	 * @param {engine.model.Document} model Model document.
```

Comment:
IMO we should call it "Document model". I know we can justify why it's model's document, but document model is a known term and reflect the reality pretty well too.